tests/test_with_defaults.py

```diff
@@ -225,7 +226,14 @@ def test_with_defaults_report_all_subtree_no_match():
     with_defaults = 'report-all'
     select = '<interfaces><interface><name>eth4</name></interface></interfaces>'
     expected = """
-<nc:data xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0"/>
+<nc:data xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0">
+  <interfaces xmlns="http://example.com/ns/interfaces">
+    <interface>
```

Comment:
should there not be a <name> node?

Reply:
Normally there would be, but in this case the result tree was made from the query with added defaults. This bypasses the normal code that would have ensured that node was present in the response.

In the test, tree is NULL, so the else is executed and tree is set to query. I think this is a rare case and it would be difficult to add node.

        if (tree)
            sch_traverse_tree (g_schema, NULL, tree, schflags | SCH_F_FILTER_RDEPTH, rdepth);
        else
        {
            /* Nothing in the database, but we may have defaults! */
            tree = query;
            query = NULL;
            sch_traverse_tree (g_schema, NULL, tree, schflags | SCH_F_FILTER_RDEPTH, rdepth);
        }

Reply:
ok .. well maybe just a note on the test that it is not actually correct and the list key "name" should also be in the output

Reply:
Done